clients/src/main/java/org/apache/kafka/common/requests/AddPartitionsToTxnRequest.java

```diff
@@ -154,6 +154,11 @@ public AddPartitionsToTxnRequest normalizeRequest() {
         return new AddPartitionsToTxnRequest(new AddPartitionsToTxnRequestData().setTransactions(singletonTransaction()), version());
     }
 
+    public boolean verifyOnlyRequest() {
```

Comment:
Let me think on this a bit -- maybe we don't need all the fields to be verify only. If I change anything, I will also add the comment explaining.

Reply:
I remember why I made this choice. We want all the requests to be verify only because the idea is that verify only requests require in memory checks only. Adding the partition otherwise requires a write to the log. 

In the case that any partition needs to be added and not just verified, we have that log write which will always take the majority of the time. In this case, it makes sense to group with the normal add partitions requests. However, in the case where all partitions are verify only, we will see faster handling and therefore want to separate the requests to different metrics as to not bring down the average for metrics like request timing.